contracts/Pair.sol

```diff
@@ -115,6 +125,24 @@ contract Pair is IPair {
         _unlocked = 1;
     }
 
+    function _safeApprove(address token, address spender, uint value) internal {
+        require(token.code.length > 0);
+        (bool success, bytes memory data) = token.call(abi.encodeWithSelector(IERC20.approve.selector, spender, value));
+        require(success && (data.length == 0 || abi.decode(data, (bool))));
+    }
+
+    function setExternalBribe(address _externalBribe) external {
```

Comment:
so theoretically this can change? should it be allowed to update only once?

Reply:
it should emit an event

Reply:
i agree this should be immutable.

Reply:
added checking with address 0

Reply:
added events